src/app/meme-card/meme-card.component.ts

```diff
@@ -17,10 +19,28 @@ export class MemeCardComponent implements OnInit {
   totalVote = 0;
   myVote = 0;
 
+  loggedInSubscription: Subscription;
+
   constructor(private memeService: MemeService,
+              private userService: UserService,
               private snackBar: MatSnackBar) { }
 
   ngOnInit() {
+    this.loggedInSubscription = this.userService.loggedIn$.subscribe((isLoggedIn) => {
+      if (isLoggedIn) {
+        // get new vote amount by refetching the meme
+        this.fetchMemeDetails();
+      } else if (this.myVote !== 0) {
```

Comment:
If we log in, we need to know if we upvoted the meme.

If we log out, our previous upvote cannot be manipulated and is then added to the total cleanly.

Reply:
but if we're not logged in, we just don't let them vote... I don't understand

Reply:
Well yes, but we have to change the totalVote counter.

Example: The meme currently has 5 votes in `totalVote` and I voted so `myVote` is 1.
In the card, it'll display the vote as `6` since my vote is not included in `totalVote`.

If I log out, `totalVote` should now be 6 since it includes votes that don't belong to me.

`otherVotes` is probably a better name, it is just the amount of votes not including you,

Reply:
What? lol... But if I voted and so it's totalVote = 6, then after I log out, I still want the vote to persist. It's just that I don't show which vote I voted for.

Reply:
Yes, the vote does persist, this just makes it so it doesn't show that you voted anymore since you logged out.

Reply:
But that doesn't make sense to me. It's fine to show my vote in the total vote. I don't recall using a program where after I voted and I log out, my vote "visually disappears". I think it's more about the state of my vote does not show up. (aka thumbs up isn't purple anymore). I think this is strange, but since the vote still persists then I'm... "ok" with it. lol

Reply:
Your vote is still counted in the same way, the UI will still show the same vote amount. The only thing that changes is the thumbs up/down.